### PR TITLE
Added UI for custom launch arguments

### PR DESCRIPTION
**Pull request by dartasen**
image

image

### CODE REVIEW

File: NitroxLauncher/Pages/ServerPage.xaml.cs [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 using System.Windows;
 using NitroxLauncher.Models;
-using NitroxLauncher.Properties;
 
 namespace NitroxLauncher.Pages
 {
     public partial class ServerPage : PageBase
     {
         public string StartButtonSubtitle => $"NITROX {LauncherLogic.RELEASE_PHASE} {LauncherLogic.Version}";
+        public bool IsServerExternal => LauncherLogic.Config.IsExternalServer;
```
Comment (Measurity) [inline NitroxLauncher/Pages/ServerPage.xaml.cs:10]:
```suggestion
        private bool IsServerExternal => LauncherLogic.Config.IsExternalServer;
```

File: NitroxLauncher/Pages/OptionPage.xaml.cs [hunk 2]
```diff
@@ -52,16 +56,36 @@ private async void ChangeOptions_Click(object sender, RoutedEventArgs e)
...
+                LaunchArguments.Text = newArguments;
+                LauncherNotifier.Success("Applied changes");
             }
             else
             {
-                MessageBox.Show("The selected directory does not contain the required Subnautica.exe file.", "Invalid Subnautica directory", MessageBoxButton.OK, MessageBoxImage.Warning);
+                LauncherNotifier.Error("Invalid launch arguments");
+                LaunchArguments.Text = SubnauticaLaunchArguments;
             }
         }
```
Comment (Measurity) [inline NitroxLauncher/Pages/OptionPage.xaml.cs:83]:
```suggestion
        private void ChangeArguments_Click(object sender, RoutedEventArgs e)
        {
            if (string.IsNullOrWhiteSpace(LaunchArguments.Text))
            {
                LaunchArguments.Text = SubnauticaLaunchArguments;
                LauncherNotifier.Error("Invalid launch arguments");
                return;
            }
            
            LaunchArguments.Text = LauncherLogic.Config.SubnauticaLaunchArguments = LaunchArguments.Text.Trim();
            LauncherNotifier.Success("Applied changes");
        }
```

**Review by Measurity — COMMENTED** [review]
As discussed, unrelated to the PR but the toast notifications do not disappear when you hover over with mouse.

**Review by Jannify — CHANGES_REQUESTED** [review]

File: NitroxLauncher/LauncherLogic.cs [hunk 3]
```diff
@@ -161,6 +161,12 @@ public void NavigateTo(Type page)
 
         internal async Task StartSingleplayerAsync()
         {
+            if (string.IsNullOrWhiteSpace(Config.SubnauticaPath) || !Directory.Exists(Config.SubnauticaPath))
+            {
+                NavigateTo<OptionPage>();
+                throw new Exception("Location of Subnautica is unknown. Set the path to it in settings.");
```
[the remark this answers is not in the document]
Comment (dartasen) [reply NitroxLauncher/LauncherLogic.cs:167]: Exceptions are shown in a ugly MessageBox like this

image

**Comment by dartasen** [comment]
Partial fix for #1657

File: NitroxLauncher/LauncherConfig.cs [hunk 4]
```diff
@@ -57,6 +58,40 @@ public string SubnauticaPath
...
+            get => subnauticaLaunchArguments;
+            set
+            {
+                if (value != subnauticaLaunchArguments)
+                {
+                    subnauticaLaunchArguments = value;
+                    Settings.Default.LaunchArgs = value;
+                    Settings.Default.Save();
+                }
+                OnPropertyChanged();
```
Comment (Measurity) [inline NitroxLauncher/LauncherConfig.cs:74]: Shouldn't property changed also be skipped if `value` is the same as current value?

File: NitroxLauncher/LauncherConfig.cs [hunk 5]
```diff
@@ -57,6 +58,40 @@ public string SubnauticaPath
...
+            get => isExternalServer;
+            set
+            {
+                if (value != isExternalServer)
+                {
+                    isExternalServer = value;
+                    Settings.Default.IsExternalServer = value;
+                    Settings.Default.Save();
+                }
+                OnPropertyChanged();
```
Comment (Measurity) [inline NitroxLauncher/LauncherConfig.cs:91]: Same here

**Review by Measurity — CHANGES_REQUESTED** [review]
Let's add a reset button for the launch arguments so people can find their way back if they mess up.

**Review by Measurity — APPROVED** [review]

**Pull request merged**